lntest/harness.go

```diff
@@ -835,15 +835,16 @@ func (n *NetworkHarness) StopNode(node *HarnessNode) error {
 
 // SaveProfilesPages hits profiles pages of all active nodes and writes it to
 // disk using a similar naming scheme as to the regular set of logs.
-func (n *NetworkHarness) SaveProfilesPages() {
+func (n *NetworkHarness) SaveProfilesPages(t *testing.T) {
 	// Only write gorutine dumps if flag is active.
 	if !(*goroutineDump) {
 		return
 	}
 
 	for _, node := range n.activeNodes {
 		if err := saveProfilesPage(node); err != nil {
-			fmt.Printf("Error: %v\n", err)
+			t.Logf("Logging follow-up error only, see rest of "+
```

Comment:
Ah interesting, so the test was actually failing due to some other reason, but this is the error that bubbled up?

Reply:
I think this was just logged first, followed by the actual cause. So we tended to overlook why the node wasn't shutting down properly in the first place.
Wee never actually called `t.Fatalf()` here, so the test couldn't fail because of an error in this loop. We just logged it in the first place. Using the test logger and adding more context will hopefully make us look at the actual cause the next time this happens.